src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -373,6 +373,9 @@ def _build_storage_profile():
     if availability_set_id:
         vm_properties['availabilitySet'] = {'id': availability_set_id}
 
+    if vmss is not None:
+        vm_properties['virtualMachineScaleSet'] = {'id': vmss}
```

Comment:
Looks like you just support `ID`

Reply:
> 
> 
> Looks like you just support `ID`

I copy the above line. I have ask this in email.

Reply:
I have supported name now.